execution_workflows/DaPars2/Dapars2_snakemake/workflow/scripts/extractIdSymbol.py

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python3
+
+import sys
+
+sys.stdout.write("#transcript_id" + '\t' + "gene_name" + "\n")
+pre_attr_id = ""
+for line in sys.stdin:
+    if not line.startswith("#"):
+        attr = dict(item.strip().split(' ') for item in line.split('\t')[8].strip('\n').split(';') if item)
+        if 'transcript_id' in attr and attr['transcript_id'] != pre_attr_id:
+            sys.stdout.write("%s\n" % (attr['transcript_id'].strip('\"') + '\t' + attr['gene_name'].strip('\"')))
+            pre_attr_id = attr['transcript_id']
```

Comment:
This works perfectly fine if the GTF is 'sorted by transcript_id' (i.e. all of the features associated with a transcript_id appear in a continuous block), otherwise there will be duplicate transcript_id gene_id rows in the output file. This shouldn't cause any issues with DaPars2 itself though as the file is loaded into a dictionary of {transcript_id: gene_id}, but if we want to be super robust this could be updated